client/embedded/pom.xml

```diff
@@ -0,0 +1,132 @@
+            <id>generate-sources</id>
+            <!-- Ensure source jar is produced before the shade plugin execution -->
+            <phase>prepare-package</phase>
+            <goals>
+              <goal>jar-no-fork</goal>
+            </goals>
+          </execution>
+          <execution>
+            <id>attach-sources</id>
+            <goals>
+              <goal>jar-no-fork</goal>
+            </goals>
+            <configuration>
+              <!-- Disable source jar configured in the parent-oss, to not override the one produced
+              by the shade plugin -->
+              <skipSource>true</skipSource>
+            </configuration>
+          </execution>
+        </executions>
+      </plugin>
+      <plugin>
+        <groupId>org.apache.maven.plugins</groupId>
+        <artifactId>maven-shade-plugin</artifactId>
+        <executions>
+          <execution>
+            <id>shade</id>
+            <phase>package</phase>
+            <goals>
+              <goal>shade</goal>
+            </goals>
+            <configuration>
+              <createDependencyReducedPom>true</createDependencyReducedPom>
+              <minimizeJar>false</minimizeJar>
+              <createSourcesJar>true</createSourcesJar>
+              <relocations>
+                <relocation>
+                  <pattern />
+                  <shadedPattern>org.sonarsource.sonarlint.core.client.shaded.</shadedPattern>
+                  <includes>
+                    <include>com.google.gson.**</include>
+                    <include>org.eclipse.lsp4j.**</include>
+                  </includes>
+                </relocation>
+              </relocations>
+              <filters>
+                <filter>
+                  <artifact>*:*</artifact>
+                  <excludes>
+                    <exclude>module-info.class</exclude>
+                    <exclude>META-INF/*.SF</exclude>
+                    <exclude>META-INF/*.DSA</exclude>
+                    <exclude>META-INF/*.RSA</exclude>
+                    <exclude>META-INF/LICENSE*</exclude>
+                    <exclude>META-INF/NOTICE*</exclude>
+                    <exclude>LICENSE*</exclude>
+                    <exclude>NOTICE*</exclude>
+                  </excludes>
+                </filter>
+                <filter>
+                  <artifact>org.sonarsource.api.plugin:sonar-plugin-api</artifact>
+                  <includes>
+                    <include>**</include>
+                  </includes>
+                  <excludes>
+                    <exclude>**/JsonWriter.class</exclude>
+                    <exclude>**/internal/google/gson/**</exclude>
+                    <!-- Overridden by SonarLint -->
+                    <exclude>org/sonar/api/utils/log/Loggers.class</exclude>
+                  </excludes>
+                </filter>
+                <filter>
+                  <!-- Shade the SonarLint Plugin API (avoids minimization) as it is used by plugins -->
```

Comment:
Both shading the API and ProtoBuf didn't work, it is not present on the artifact.

Reply:
OK, I will fix that in another PR